unblob/handlers/archive/ewf/ewf.py

```diff
@@ -0,0 +1,202 @@
+
+    typedef struct volume_descriptor {
+        uint32 unknown;
+        uint32 chunk_count;
+        uint32 sectors_per_chunks;
+        uint32 bytes_per_sectors;
+        uint32 sectors_count;
+    } volume_descriptor_t;
+
+    typedef struct table_descriptor {
+        uint32 number_of_entries;
+        char padding[16];
+        uint32 adler_32;
+    } table_descriptor_t;
+
+    typedef struct table_entry {
+        char offset[3];
+        char compression_type[1];
+    } table_entry_t;
+    typedef struct hash_descriptor {
+        char md5_hash[16];
+        char unknown[16];
+        uint32 adler_32;
+    } hash_descriptor_t;
+"""
+
+EWF_HEADER_LEN = 13
+DESCRIPTOR_LEN = 76
+
+
+class Definition(Enum):
+    DONE = b"done".ljust(16, b"\x00")
+    TABLE = b"table".ljust(16, b"\x00")
+    SECTORS = b"sectors".ljust(16, b"\x00")
+    VOLUME = b"volume".ljust(16, b"\x00")
+
+
+class ZlibMagic(Enum):
+    LOW = b"\x78\x01"
+    DEFAULT = b"\x78\x9c"
+    BEST = b"\x78\xda"
+    COMPRESSION = b"\x78\x5e"
+
+
+def find_chunk_size(header: Instance) -> int:
+    size = header.sectors_per_chunks
+    log = math.log(size, 2)
+    power = math.pow(2, log + 9)
+    return int(power)
+
+
+def is_valid_header(header: Instance) -> bool:
+    if header.field_start != 0x01 or header.field_end != 0x0:
+        return False
+    return True
+
+
+class EWFExtractor(Extractor):
+    def __init__(self, header_struct: str):
+        self.header_struct = header_struct
+        self._struct_parser = StructParser(C_DEFINITIONS)
+
+    def table_descriptor(
+        self, file: File, position: int, outdir: Path, sectors_per_chunk: int
+    ) -> Instance:
+        fs = FileSystem(outdir)
+        entries = []
+        header = self._struct_parser.parse("table_descriptor_t", file, Endian.LITTLE)
+        entry_path = Path("ewf.decrypted")
+
+        for _ in range(header.number_of_entries):
+            entry = self._struct_parser.parse("table_entry_t", file, Endian.LITTLE)
+            entries.append(entry.offset)
+
+        with fs.open(entry_path) as output_file:
+            for offset in entries:
+                file.seek(
+                    position
+                    + int.from_bytes(offset, byteorder="little")
+                    - DESCRIPTOR_LEN,
+                    io.SEEK_SET,
+                )
+
+                magic_bytes = file.read(2)
+                compressed = any(magic_bytes == magic.value for magic in ZlibMagic)
+
+                for chunk in iterate_file(
+                    file,
+                    position
+                    + int.from_bytes(offset, byteorder="little")
+                    - DESCRIPTOR_LEN,
+                    sectors_per_chunk,
+                ):
+                    if compressed:
+                        compressed_chunk = zlib.decompress(chunk)
+                        output_file.write(compressed_chunk)
+                    output_file.write(chunk)
+
+    def extract(self, inpath: Path, outdir: Path):
```

Comment:
There is a logic / naming issue here. The **extraction** is performed in `table_descriptor` while the **parsing** is performed in `extract`. Please adjust your function so they reflect what they're actually doing. Ideally you would parse the EWF structure and return data that can be used by `extract` in order to create files in `outdir`.